app/src/main/java/org/mozilla/rocket/download/DownloadInfoRepository.kt

```diff
@@ -0,0 +1,71 @@
+package org.mozilla.rocket.download
+
+import android.app.Application
+import android.arch.lifecycle.LiveData
+import org.mozilla.focus.download.DownloadInfo
+import org.mozilla.focus.download.DownloadInfoManager
+
+class DownloadInfoRepository {
+
+    var downloadIndicator: DownloadIndicatorLiveData? = null
+    var downloadInfoBundle: DownloadInfoLiveData? = null
+
+    fun getDownloadIndicator(): LiveData<Int>? {
+        return downloadIndicator
+    }
+
+    fun getDownloadInfoBundle(): LiveData<DownloadInfoBundle>? {
+        return downloadInfoBundle
+    }
+
+    fun cancelDownload(id: Long, listener: DownloadInfoViewModel.OnOperationComplete) {
+        downloadInfoBundle?.cancel(id, listener)
+    }
+
+    fun deleteDownload(id: Long, listener: DownloadInfoViewModel.OnOperationComplete) {
+        downloadInfoBundle?.delete(id, listener)
+    }
+
+    fun removeDownload(id: Long) {
+        downloadInfoBundle?.removeItem(id)
+    }
+
+    fun hideDownload(id: Long) {
+        downloadInfoBundle?.hideItem(id)
+    }
+
+    fun addDownload(downloadInfo: DownloadInfo) {
+        downloadInfoBundle?.addItem(downloadInfo)
+    }
+
+    fun loadMore(init: Boolean) {
+        downloadInfoBundle?.loadMore(init)
+    }
+
+    fun isOpening(): Boolean? {
+        return downloadInfoBundle?.isOpening
+    }
+
+    fun setOpening(value: Boolean) {
+        downloadInfoBundle?.isOpening = value
+    }
+
+    fun markAllItemsAreRead() {
+        DownloadInfoManager.getInstance().markAllItemsAreRead(null)
+    }
+
+    companion object {
+
+        @Volatile private var INSTANCE: DownloadInfoRepository? = null
+
+        @JvmStatic
+        fun getInstance(application: Application): DownloadInfoRepository? =
```

Comment:
why we need the application here?  If we want that live data, can we use transformation instead?

Reply:
See: https://medium.com/androiddevelopers/viewmodels-and-livedata-patterns-antipatterns-21efaef74a54